namesys/publisher.go

```diff
@@ -105,6 +106,11 @@ func ipnsEntryDataForSig(e *pb.IpnsEntry) []byte {
 		[]byte{})
 }
 
+var IpnsRecordValidator = &record.ValidChecker{
+	Func: ValidateIpnsRecord,
+	Sign: true,
```

Comment:
@whyrusleeping it's fine for some of them to be impure-- time is impure. either way, i'd like to move to just functions, but not a huge deal atm.